From ea52f21dfd8eb0022911483412cc9e73d4365dcf Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 2 Dec 2021 10:05:12 +1100
Subject: [PATCH 02/12] ensure we set PROTOBUF_USE_DLLS when using our own
 protobuf

---
 src/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7156b3f..ebba293 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -68,6 +68,11 @@ if (SPM_PROTOBUF_PROVIDER STREQUAL "internal")
   include_directories(builtin_pb)
 elseif (SPM_PROTOBUF_PROVIDER STREQUAL "package")
   find_package(Protobuf REQUIRED)
+  if (MSVC)
+   add_definitions("/DPROTOBUF_USE_DLLS")
+  else()
+   add_definitions("-DPROTOBUF_USE_DLLS")
+  endif()
   include_directories(${Protobuf_INCLUDE_DIRS})
   protobuf_generate_cpp(SPM_PROTO_SRCS SPM_PROTO_HDRS sentencepiece.proto)
   protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS sentencepiece_model.proto)
