From 6505b729e8678d7f58fc73f4adb821586792226d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Nov 2024 11:49:46 +1100
Subject: [PATCH 07/12] respect SPM_ENABLE_SHARED also on windows

---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93fbac9..6f3987f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,11 +58,6 @@ if (SPM_CROSS_SYSTEM_PROCESSOR)
  set(CMAKE_SYSTEM_PROCESSOR ${SPM_CROSS_SYSTEM_PROCESSOR})
 endif()
 
-# Disable shared build on windows
-if(WIN32)
-  set(SPM_ENABLE_SHARED OFF)
-endif()
-
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
