From f676c070107caaff2cd3315a08e67d438879f2f4 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Nov 2024 12:33:34 +1100
Subject: [PATCH 08/12] add SPM_DLL machinery

---
 src/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fe6cc6b..23cf9d9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -74,8 +74,10 @@ elseif (SPM_PROTOBUF_PROVIDER STREQUAL "package")
    add_definitions("-DPROTOBUF_USE_DLLS")
   endif()
   include_directories(${Protobuf_INCLUDE_DIRS})
-  protobuf_generate_cpp(SPM_PROTO_SRCS SPM_PROTO_HDRS sentencepiece.proto)
-  protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS sentencepiece_model.proto)
+  protobuf_generate_cpp(SPM_PROTO_SRCS SPM_PROTO_HDRS
+    EXPORT_MACRO SPM_DLL sentencepiece.proto)
+  protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS
+    EXPORT_MACRO SPM_DLL sentencepiece_model.proto)
   set(PROTOBUF_LITE_SRCS "")
   include_directories(${PROTOBUF_INCLUDE_DIR})
   add_compile_definitions(_USE_EXTERNAL_PROTOBUF)
@@ -254,6 +256,8 @@ if (SPM_ENABLE_SHARED)
     set_target_properties(sentencepiece PROPERTIES IMPORT_SUFFIX ".dll.a")
     set_target_properties(sentencepiece_train PROPERTIES IMPORT_SUFFIX ".dll.a")
   endif()
+  set_target_properties(sentencepiece PROPERTIES DEFINE_SYMBOL "SPM_DLL_EXPORTS")
+  target_compile_definitions(sentencepiece INTERFACE SPM_DLL_IMPORTS)
 else()
   add_library(sentencepiece ALIAS sentencepiece-static)
   add_library(sentencepiece_train ALIAS sentencepiece_train-static)
